### PR TITLE
feat: index anchor and nonce in Profile allov2

### PR DESCRIPTION
**Pull request by thelostone-mc**
image

Co-authored-by: 0xKurt <Kurt@Merbeth.io>

fixes https://github.com/gitcoinco/grants-stack-indexer/issues/424

### CODE REVIEW

**Comment by boudra** [comment]
@codenamejason the schema graphql file is ignored now but it has to be deleted from git `git rm`

File: src/database/schema.ts [hunk 1]
```diff
@@ -75,6 +75,8 @@ export type PartialRoundRole = Updateable<RoundRoleTable>;
 export type ProjectTable = {
   id: string;
   name: string;
+  nonce: bigint | null;
+  anchor: Address | null;
```
Comment (boudra) [inline src/database/schema.ts:79]:
```suggestion
  anchorAddress: Address | null;
```

it has to be modified everywhere else 👀

**Review by boudra — CHANGES_REQUESTED** [review]

File: src/indexer/allo/v2/handleEvent.ts [hunk 2]
```diff
@@ -55,6 +55,8 @@ export async function handleEvent(
             registryAddress: parseAddress(event.address),
             id: profileId,
             name: event.params.name,
+            nonce: event.params.nonce,
+            anchor: parseAddress(event.params.anchor),
```
Comment (thelostone-mc) [inline src/indexer/allo/v2/handleEvent.ts:59]:
```suggestion
            anchorAddress: parseAddress(event.params.anchor),
```

File: src/indexer/allo/v2/handleEvent.test.ts [hunk 3]
```diff
@@ -422,6 +428,8 @@ describe("handleEvent", () => {
         name: "",
         tags: ["allo-v2"],
         chainId: 1,
+        nonce: 1n,
+        anchor: parseAddress(addressThree),
```
Comment (thelostone-mc) [inline src/indexer/allo/v2/handleEvent.test.ts:432]:
```suggestion
        anchorAddress: parseAddress(addressThree),
```

File: src/indexer/allo/v2/handleEvent.test.ts [hunk 4]
```diff
@@ -346,6 +350,8 @@ describe("handleEvent", () => {
           name: "",
           tags: ["allo-v2"],
           chainId: 1,
+          nonce: 1n,
+          anchor: parseAddress(addressZero),
```
Comment (thelostone-mc) [inline src/indexer/allo/v2/handleEvent.test.ts:354]:
```suggestion
          anchorAddress: parseAddress(addressZero),
```

File: src/indexer/allo/v2/handleEvent.test.ts [hunk 5]
```diff
@@ -186,6 +188,8 @@ describe("handleEvent", () => {
           project: {
             chainId: 1,
             name: "Project 1",
+            nonce: 1n,
+            anchor: addressThree,
```
Comment (thelostone-mc) [inline src/indexer/allo/v2/handleEvent.test.ts:192]:
```suggestion
            anchorAddress: addressThree,
```

File: src/indexer/allo/v2/handleEvent.test.ts [hunk 6]
```diff
@@ -111,6 +111,8 @@ describe("handleEvent", () => {
...
```
Comment (thelostone-mc) [inline src/indexer/allo/v2/handleEvent.test.ts:115]:
```suggestion
            anchorAddress: addressThree,
```

**Review by boudra — APPROVED** [review]

**Pull request merged**